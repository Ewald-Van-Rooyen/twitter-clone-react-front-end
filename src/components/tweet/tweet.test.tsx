import TestRenderer, {ReactTestRenderer} from "react-test-renderer";
import Tweet from "./tweet";
import React from "react";

let testRenderer: ReactTestRenderer;
const fullname = "Ewald Van Rooyen";
const username = "@Ewald42";
const date = "2015-01-17";
const claps = 50;
const profilePic = "https://avatarfiles.alphacoders.com/174/174225.jpg";
const tweet = "Hello World";

describe("Tweet component tests", () => {

    beforeAll(() => {
        testRenderer = TestRenderer.create(
            <Tweet date={date}
                   tweet={tweet}
                   claps={claps}
                   id={1}
                   userId={2}/>
        );
    });

    it("should render the Tweet component", () => {
        expect(testRenderer).toMatchSnapshot();
    });

    it("should match the props", () => {
        const testInstance = testRenderer.root;

        expect(testInstance.props.fullname).toBe(fullname);
        expect(testInstance.props.username).toBe(username);
        expect(testInstance.props.date).toBe(date);
        expect(testInstance.props.claps).toBe(claps);
        expect(testInstance.props.profilePic).toBe(profilePic);
        expect(testInstance.props.tweet).toBe(tweet);
    });

    it("should check for class names", () => {
        const testInstance = testRenderer.root;
        const div = testInstance.findByType("div");
        const svg = testInstance.findByType("svg");
        const img = testInstance.findByType("img");

        expect(div.props.className.includes("tweet-container")).toBe(true);
        expect(img.props.className.includes("avatar")).toBe(true);
        expect(svg.props.className.includes("bi bi-heart")).toBe(true);
    });

});


