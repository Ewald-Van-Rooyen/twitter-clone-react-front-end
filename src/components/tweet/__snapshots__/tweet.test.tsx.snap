// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet component should render the Tweet component 1`] = `
<div
  className="tweet-container"
>
  <div
    className="tweet-header"
  >
    <img
      alt=""
      className="avatar"
      src="https://avatarfiles.alphacoders.com/174/174225.jpg"
    />
    <div
      className="tweet-header-info"
    >
      Ewald Van Rooyen
      <span>
        @Ewald42
      </span>
      <span>
        2015-01-17
      </span>
      <p>
         
        Hello World
      </p>
    </div>
  </div>
  <div
    className="tweet-icons"
  >
    <div
      className="likes"
    >
      <svg
        className="bi bi-heart"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
        />
      </svg>
      <div
        className="likes-count"
      >
        50
      </div>
    </div>
  </div>
</div>
`;

exports[`Tweet component tests should render the Tweet component 1`] = `
<div
  className="tweet-container"
>
  <div
    className="tweet-header"
  >
    <img
      alt=""
      className="avatar"
      src="https://avatarfiles.alphacoders.com/174/174225.jpg"
    />
    <div
      className="tweet-header-info"
    >
      Ewald Van Rooyen
      <span>
        @Ewald42
      </span>
      <span>
        2015-01-17
      </span>
      <p>
         
        Hello World
      </p>
    </div>
  </div>
  <div
    className="tweet-icons"
  >
    <div
      className="likes"
    >
      <svg
        className="bi bi-heart"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
        />
      </svg>
      <div
        className="likes-count"
      >
        50
      </div>
    </div>
  </div>
</div>
`;
